### ISSUE TITLE
Evaluate the network on the validation set during training. Save the weights for the best performance on validation set

### ISSUE THREAD

**Issue by SergeyHayrapetyaNN** [pjreddie/darknet#782]
Able to calculate the loss, avg_iou, ap, etc on the validation set

**Comment by mittlin** [comment]
@SergeyHayrapetyaNN   any idea??
